Fw/Types/ObjectName.cpp

```diff
@@ -0,0 +1,55 @@
+#include <Fw/Types/ObjectName.hpp>
+#include <Fw/Types/StringUtils.hpp>
+
+namespace Fw {
+
+    ObjectName::ObjectName(const CHAR* src) : StringBase() {
+        Fw::StringUtils::string_copy(this->m_buf, src, sizeof(this->m_buf));
```

Comment:
Check src is not nullptr? Or is string_copy graceful at that?

Reply:
I think this check should be done at the point of use, inside the function body. It looks like the check is not currently done though.

https://github.com/nasa/fprime/blob/58c21418aac3162156299ad6e0153a1e94288b45/Fw/Types/StringUtils.cpp#L5-L18

Reply:
It would be good to add the null pointer check for source and destination before calling the library function line 15.

Reply:
Added the check in string_copy as suggested